src/cuda/cuda-utils.cpp

```
@@ -81,59 +83,35 @@ void checkCudaSyncError(bool pre, const char* call, const char* file, int line)
-        {
-            device->handleMessage(
-                DebugMessageType::Error,
-                DebugMessageSource::Driver,
-                "Error detected BEFORE the call, suggesting a prior, uncaptured CUDA call is responsible\n"
-            );
-        }
+        reportCUDAError(result, call, location, device);
+        const char* message =
+            pre ? "Error detected BEFORE the call, suggesting a prior, uncaptured CUDA call is responsible\n"
+                : "Error detected AFTER the call, suggesting it is responsible\n";
+        if (device)
```

Cppcheck (2.20.0)

[style] The function 'clamp' is never used.

(unusedFunction)